server/container_create.go

```diff
@@ -201,6 +201,9 @@ func setupContainerUser(ctx context.Context, specgen *generate.Generator, rootfs
 	for _, env := range specgen.Config.Process.Env {
 		if strings.HasPrefix(env, "HOME=") {
 			homedir = strings.TrimPrefix(env, "HOME=")
+			if idx := strings.Index(homedir, `\n`); idx > -1 {
+				return fmt.Errorf("invalid HOME environment; newline not allowed")
```

Comment:
Instead of failing, can we continue by dropping `\n`  from the variable?

Reply:
if the value is set to a path with a newline because the path genuinely has a newline, then the HOME won't be set and the container will fail more nefariously. I think it'd be better to fail quickly and notify the user why, rather than let the homedir be set wrong

Reply:
Oh ok, makes sense. Thanks.